### ISSUE TITLE
There is a problem with FoundationStereo training.

### ISSUE THREAD

**Issue by ChatGPTJHBHJH** [XiandaGuo/OpenStereo#264]
Hello, author. I would like to ask you a question regarding the DepthAnything branch in FoundationStereo.

<img width="503" height="43" alt="Image" src="https://github.com/user-attachments/assets/b60c9797-dc63-4d97-8dc1-aa302eea0060" />

The weights of DepthAnythingFeature are frozen, but it seems that only dinov2 is loaded when loading the weights. The weights of self.depth_head = DPTHead(1, dim, features, use_bn, out_channels=out_channels, use_clstoken=use_clstoken) do not seem to be loaded before freezing. As a result, the performance of my own training is very poor. Is there something wrong with my observation? It seems that self.depth_head should load the pre-trained weights before being frozen. Hope you can reply, thank you very much!

<img width="1192" height="160" alt="Image" src="https://github.com/user-attachments/assets/de6a87c2-f8b5-4aad-8632-ff3322d616e0" />

**Comment by xpyqiubai** [comment]
I have the same question about this. pretrained_dino=False. I think it's likely that none of the pre-trained parameters were loaded. Have you found a solution to this problem?

**Comment by XiandaGuo** [comment]
Thanks for your feedback! We have reviewed the code, and the current model-related implementation is consistent with the original FoundationStereo (https://github.com/NVlabs/FoundationStereo).
<img width="1752" height="540" alt="Image" src="https://github.com/user-attachments/assets/f47938bb-f846-445b-a88d-2e64b8f0a430" />
However, we’ve verified the issue you raised and made the suggested code modifications (see the attached screenshot for reference). The final results did not show improvement. We trained the model for 50 epochs on SceneFlow, with the following outcomes:
Original code: EPE = 0.5421
Modified code: EPE = 0.5547
For further clarification, I recommend reaching out to the original authors directly via the official FoundationStereo(https://github.com/NVlabs/FoundationStereo) repository. 
Feel free to add me on WeChat (2357978449) if you’d like to discuss this in more detail!
